subsys/logging/log_output_custom.c

```diff
@@ -20,3 +23,25 @@ void log_custom_output_msg_set(log_format_func_t format)
 {
 	log_custom_format_func = format;
 }
+#endif /* defined(CONFIG_LOG_CUSTOM_FORMAT_SUPPORT) */
+
+#if defined(CONFIG_LOG_OUTPUT_FORMAT_CUSTOM_TIMESTAMP)
```

Comment:
can you check if removing those ifdefs impacts the code size? i think that there might be a linker garbage collection and function is removed when not used.

Reply:
You are correct. I removed the `#if`s and the code size doesn't change.